web/src/Documentation.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect } from 'react'
+import {
+    useParams, Link
+} from "react-router-dom";
+
+import Markdown from 'react-markdown'
+import remarkGfm from 'remark-gfm'
+import remarkToc from 'remark-toc'
+import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter'
+// import { materialDark as syntaxStyle } from 'react-syntax-highlighter/dist/esm/styles/prism'
+
+interface IDocumentationArticleProps {
+    articleId?: string
+}
+
+function RouterLink(props: any) {
+    if (props.href.match(/^(https?:)?\/\//))
+        return <a href={props.href}>{props.children}</a>
+
+
+    let link: string = props.href
+    if (!link.startsWith('/')) { link = '/documentation/' + link; }
+
+    return <Link to={link}>{props.children}</Link>;
+}
+
+const DocumentationArticle: React.FunctionComponent<IDocumentationArticleProps> = (props: IDocumentationArticleProps) => {
+
+    const [text, setText] = React.useState<string | null>(null)
+
+    const match = useParams()
+
+    useEffect(() => {
+        console.log('Loading data')
```

Comment:
@danielreti 
The `useEffect` here with no dependencies should act like `componentDidMount` or `componentDidUpdate` functions, allowing me to load data on first load, or reload if the props change. An optimisation here could be to only reload if the articleId changes, but not required at the moment,